contracts/cw2981-royalties/src/query.rs

```diff
@@ -0,0 +1,42 @@
+use crate::msg::{CheckRoyaltiesResponse, RoyaltiesInfoResponse};
+use crate::Cw2981Contract;
+use cosmwasm_std::{Decimal, Deps, StdResult, Uint128};
+
+// NOTE: default behaviour here is to round down
+// EIP2981 specifies that the rounding behaviour is at the discretion of the implementer
+pub fn query_royalties_info(
+    deps: Deps,
+    token_id: String,
+    sale_price: Uint128,
+) -> StdResult<RoyaltiesInfoResponse> {
+    let contract = Cw2981Contract::default();
+    let token_info = contract.tokens.load(deps.storage, &token_id)?;
+
+    let royalty_percentage = match token_info.extension {
+        Some(ref ext) => match ext.royalty_percentage {
+            Some(percentage) => Decimal::percent(percentage),
+            None => Decimal::percent(0),
+        },
+        None => Decimal::percent(0),
+    };
+    let royalty_from_sale_price = sale_price * royalty_percentage;
+
+    let royalty_address = match token_info.extension {
+        Some(ext) => match ext.royalty_payment_address {
+            Some(addr) => addr,
+            None => String::from(""),
+        },
+        None => String::from(""),
+    };
+
+    Ok(RoyaltiesInfoResponse {
+        address: royalty_address,
+        royalty_amount: royalty_from_sale_price,
+    })
+}
+
+pub fn check_royalties(_deps: Deps) -> StdResult<CheckRoyaltiesResponse> {
+    Ok(CheckRoyaltiesResponse {
+        royalty_payments: true,
```

Comment:
Wait, always true?

What about `ext.royalty_percentage.is_some()`

Reply:
Me being a little lazy as it's an example contract - it's a check at contract level, for this one the answer to "should I see if, when a token controlled by this contract is sold, royalties are owed?" will always be yes. I think `ext.royalty_percentage.is_some()` would be relevant if you were querying on the `token_id`. Then again, this is based on my interpretation of the granularity of the EIP spec, so I could have it wrong 🙂

Reply:
Seems like this is a case where we don't need to follow Ethereum exactly. The boolean seems redundant.

Reply:
No, it will still return boolean regardless - that's the interface, I'm just saying that `ext.royalty_percentage.is_some()` is at a different level of granularity - token, vs contract

Reply:
Rechecked the spec and this interface needs to return a boolean, so yeah the answer to:

>  Checking if the NFT being sold on your marketplace implemented royalties 

For a function that takes a contract and returns a boolean - this _is_ the correct base case, and then if contracts want to implement custom logic, they can. This is just the most basic correct example, which is that when token is sold, it signals you should call `query_royalties_info` with the token ID.

Reply:
Still not a big fan of the `CheckRoyalties` query as I think it's mostly pointless. It's more important in the context of Ethereum.

Could just query `RoyaltiesInfo` and see if it returns a response for the same effect.

Reply:
@orkunkl, any thoughts on this?